File: basic recursion/javascript/n_times.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function repeatString(str, n) {
    if (n <= 0) {
        return "";
    }
    return str + repeatString(str, n - 1);
}

rl.question("Enter a string: ", function(str) {
    rl.question("Enter the number of times to repeat: ", function(n) {
        n = parseInt(n);
        if (isNaN(n) || n <= 0) {
            console.log("Please enter a valid positive integer.");
        } else {
            const result = repeatString(str, n);
            console.log(`The string "${str}" repeated ${n} times is:`)
            console.log(result);
        }
        rl.close();
    });
});

File: basic recursion/c++/n_times.cpp
#include <iostream>
using namespace std;

void printName(int n, string name)
{
    if (n == 0)
    {
        return;
    }
    cout << name << " ";
    printName(n - 1, name);
}
int main()
{
    int n;
    string name;
    cout << "Enter a number: ";
    cin >> n;
    cout << "Enter a name: ";
    cin >> name;

    printName(n, name);
    return 0;
}